arches/app/utils/geo_utils.py

```diff
@@ -94,3 +99,142 @@ def convert_geos_geom_collection_to_feature_collection(self, geometry):
             arches_geojson["features"].append(arches_json_geometry)
         return arches_geojson
+
+    def get_resource_instances_within_feature_collection(self, feature_collection):
+        """
+        Takes a FeatureCollection object and returns a dictionary of resource instances
+        that intersect the geometries of it, grouped by graph.
+        """
+        with connection.cursor() as cursor:
+            cursor.execute(
+                """
+                WITH combined_geom AS (
+                    SELECT ST_Union(
+                        ST_Transform(
+                            ST_SetSRID(ST_GeomFromGeoJSON(feature->>'geometry'), 4326), 
+                            3857
+                        )
+                    ) AS geom
+                    FROM jsonb_array_elements(%s::jsonb->'features') AS feature
+                )
+                SELECT resource_instances.graphid, 
+                    array_agg(geojson_geometries.resourceinstanceid) AS resourceinstanceids
+                FROM geojson_geometries 
+                JOIN resource_instances 
+                ON geojson_geometries.resourceinstanceid = resource_instances.resourceinstanceid
+                WHERE ST_Intersects(
+                        geojson_geometries.geom, 
+                        (SELECT geom FROM combined_geom)
+                    )
+                AND resource_instances.graphid != %s
+                GROUP BY resource_instances.graphid;
+                """,
+                [
+                    json.dumps(feature_collection),
+                    settings.SYSTEM_SETTINGS_RESOURCE_MODEL_ID,
+                ],
+            )
+
+            results = cursor.fetchall()
+
+        return dict(results)
+
+    def buffer_feature_collection(self, feature_collection, buffer_distance):
+        """
+        Takes a FeatureCollection object and a value for the buffer distance in meters,
+        and returns a FeatureCollection of the original features with the buffer distance added.
+
+        """
+
+        buffered_features = []
+
+        for feature in feature_collection["features"]:
+            geom = GEOSGeometry(json.dumps(feature["geometry"]))
+
+            geom.transform(settings.ANALYSIS_COORDINATE_SYSTEM_SRID)
+            buffered_geom = geom.buffer(buffer_distance)
+            buffered_geom.transform(4326)
+
+            buffered_feature = {
+                "type": "Feature",
+                "id": str(uuid.uuid4()),
+                "properties": feature.get("properties", {}),
+                "geometry": json.loads(buffered_geom.geojson),
+            }
+            buffered_features.append(buffered_feature)
+
+        return {
+            "type": "FeatureCollection",
+            "features": buffered_features,
+        }
+
+    def get_intersection_and_difference_of_feature_collections(
+        self, feature_collection_1, feature_collection_2
+    ):
+        def _build_feature_collection(geometries):
```

Comment:
a couple thoughts here:
- `geometries` appears to be a single GEOS geometry, so I think we should rename it to `geometry`
- is there a reason for decomposing multi-part geometries before creating features?  those are valid geojson geometry types, so there doesn't seem to be a strict requirement for that in my mind.